File: Excel/673 Challenge.R
library(tidyverse)
library(readxl)
library(matricks)

path = "Excel/673 Star Pattern.xlsx"
test3  = read_excel(path, range = "C3:E5", col_names = FALSE) %>% as.matrix()
test4  = read_excel(path, range = "C7:F10", col_names = FALSE) %>% as.matrix()
test7 = read_excel(path, range = "C12:I18", col_names = FALSE) %>% as.matrix()

make_star = function(side) {
  M = matrix(NA_character_, side, side)
  for (i in 1:side){
    M[i,i] = LETTERS[i]
    M[i,side+1-i] = LETTERS[i]
  } 
  return(M)
}

all.equal(make_star(3), test3, check.attributes = FALSE) # TRUE
all.equal(make_star(4), test4, check.attributes = FALSE) # TRUE
all.equal(make_star(7), test7, check.attributes = FALSE) # TRUE

File: Excel/600-699/673/673 Challenge.py
import pandas as pd
import numpy as np

path = "673 Star Pattern.xlsx"

test3 = pd.read_excel(path, usecols="C:E", skiprows=2, nrows=3, header=None).fillna('').values
test4 = pd.read_excel(path, usecols="C:F", skiprows=6, nrows=4, header=None).fillna('').values
test7 = pd.read_excel(path, usecols="C:I", skiprows=11, nrows=7, header=None).fillna('').values

def make_star(side):
    M = np.full((side, side), '', dtype=str)
    M = np.array([[chr(65 + i) if j == i or j == side - 1 - i else '' for j in range(side)] for i in range(side)])
    return M

print(np.array_equal(make_star(3), test3))  # True
print(np.array_equal(make_star(4), test4))  # True
print(np.array_equal(make_star(7), test7))  # True
